### PR TITLE
More accurate error messages

### PR DESCRIPTION
**Pull request by KvanTTT**

### CODE REVIEW

File: tool-testsuite/test/org/antlr/v4/test/tool/Java.g4 [hunk 1]
```diff
@@ -1140,7 +1140,7 @@ SignedInteger
 
 fragment
 Sign
-	:	[+-]
+	:	[+\-]
```
Comment (mike-lischke) [inline tool-testsuite/test/org/antlr/v4/test/tool/Java.g4]: So this is no longer possible without escaping the minus? It worked when the minus was at the end of the set list.

Comment (KvanTTT) [inline tool-testsuite/test/org/antlr/v4/test/tool/Java.g4]: Yes, not possible. Old syntax was a bit incorrect.

File: tool/src/org/antlr/v4/automata/ATNOptimizer.java [hunk 2]
```diff
@@ -147,6 +150,29 @@ private static void optimizeSets(Grammar g, ATN atn) {
 //		System.out.println("ATN optimizer removed " + removedStates + " states by collapsing sets.");
 	}
 
+	private static String getIntervalSetEscapedString(IntervalSet intervalSet) {
```
Comment (parrt) [inline tool/src/org/antlr/v4/automata/ATNOptimizer.java]: not sure this belongs in this class

Comment (KvanTTT) [inline tool/src/org/antlr/v4/automata/ATNOptimizer.java]: Ok, I'll move it to `CharSupport`.

**Comment by KvanTTT** [comment]
Much better :)

File: tool-testsuite/test/org/antlr/v4/test/tool/Java.g4 (hunk 1 above)
Comment (sharwell) [inline tool-testsuite/test/org/antlr/v4/test/tool/Java.g4]: @KvanTTT Old syntax was pretty common. I expect it to remain valid.

**Comment by parrt** [comment]
@sharwell oh, yeah, didn't we long time ago say anything not terminated was just the '-' char? Dang. @KvanTTT how big is the change to flip `a-` and `-z` to be two char not a range.

**Comment by KvanTTT** [comment]
Pretty easy.

**Comment by parrt** [comment]
ok, let's do that. thanks!

**Pull request merged**